### PR TITLE
Fix configurations for UAP

### PR DESCRIPTION
**Pull request by ericstj**
These changes enable TargetGroup=UAP to build completely.

@joperezr you should be able to work from here and just add configurations to projects that aren't building.

/cc @weshaggard

### CODE REVIEW

File: src/System.Diagnostics.FileVersionInfo/src/Configurations.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <BuildConfigurations>
-      netcore50-Windows_NT;
+      uap-Windows_NT;
```
Comment (joperezr) [inline src/System.Diagnostics.FileVersionInfo/src/Configurations.props]: Is the idea to always replace netcore50 configs for uap ones? Or only in the case they don't have a pkgproj like this one?

Comment (ericstj) [inline src/System.Diagnostics.FileVersionInfo/src/Configurations.props]: I just did the minimal set needed to build.  It's possible there is more cleanup to do.

You are correct that projects without packages don't need any "old" configurations.  Also for projects with packages we can remove old configurations and opt to harvest those assemblies from the previous package.

**Comment by joperezr** [comment]
>       <CompatibleWith>netstandard2.0</CompatibleWith>

I can add it as part of my PR, but the way I had uap101aot set up is:
```xml
    <TargetGroups Include="uap101aot">
      <PackageTargetRuntime>aot</PackageTargetRuntime>
      <NuGetTargetMoniker>UAP,Version=v10.1</NuGetTargetMoniker>
      <Imports>netcore50aot</Imports>
      <CompatibleWith>uap101;netstandard2.0</CompatibleWith>
    </TargetGroups>
```

---
Refers to: src/Tools/GenerateProps/targetgroups.props:20 in 17fdb43. [](commit_id = 17fdb439c50f4aa66a99f9b5549e9db923331e40, deletion_comment = False)

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj [hunk 2]
```diff
@@ -5,14 +5,14 @@
     <AssemblyName>System.Net.Sockets</AssemblyName>
     <ProjectGuid>{43311AFB-D7C4-4E5A-B1DE-855407F90D1B}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <EnableWinRT Condition="'$(TargetGroup)' == 'netcore50'">true</EnableWinRT>
+    <EnableWinRT Condition="'$(TargetGroup)' == 'uap'">true</EnableWinRT>
     <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'net463'">true</IsPartialFacadeAssembly>
-    <AssemblyVersion Condition="'$(TargetGroup)' == 'netcore50'">4.1.1.0</AssemblyVersion>
-    <DefineConstants Condition="'$(TargetGroup)' == 'netcore50'">$(DefineConstants);netcore50</DefineConstants>
+    <AssemblyVersion Condition="'$(TargetGroup)' == 'uap'">4.1.1.0</AssemblyVersion>
```
Comment (weshaggard) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: We probably don't need these special assembly versions any longer.

Comment (ericstj) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Yeah, I deleted where I noticed it but probably missed a few.  @joperezr feel free to clean up in your work.

**Pull request merged**